test/awsE2ESolutionSetup.sh

```diff
@@ -0,0 +1,152 @@
+bootstrap_region () {
+  # Picking arbitrary context values to satisfy required values for CDK synthesis. These should not need to be kept in sync with the actual deployment context values
+  cdk bootstrap "*" --require-approval never --c suffix="Migration-Source" --c distVersion="7.10.2" --c distributionUrl="https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-oss-7.10.2-linux-x86_64.tar.gz" --c securityDisabled=true --c minDistribution=false --c cpuArch="x64" --c isInternal=true --c singleNodeCluster=false --c networkAvailabilityZones=2 --c dataNodeCount=2 --c managerNodeCount=0 --c serverAccessType="ipv4" --c restrictServerAccessTo="0.0.0.0/0" --c captureProxyEnabled=false
+}
+
+usage() {
+  echo ""
+  echo "Script to setup E2E AWS infrastructure for an ES 7.10.2 source cluster running on EC2, as well as "
+  echo "an OpenSearch Service Domain as the target cluster, and the Migration infrastructure for simulating a migration from source to target."
+  echo ""
+  echo "Usage: "
+  echo "  ./awsE2ESolutionSetup.sh [--create-service-linked-roles] [--bootstrap-region] [--enable-capture-proxy]"
+  echo ""
+  echo "Options:"
+  echo "  --create-service-linked-roles                    If included, will attempt to create required service linked roles for the AWS account"
+  echo "  --bootstrap-region                               If included, will attempt to CDK bootstrap the region to allow CDK deployments"
+  echo "  --enable-capture-proxy                           If included, will attempt to enable the capture proxy on all source nodes"
+  echo ""
+  exit 1
+}
+
+stage='aws-integ'
+CREATE_SLR=false
+BOOTSTRAP_REGION=false
+ENABLE_CAPTURE_PROXY=false
+while [[ $# -gt 0 ]]; do
+  case $1 in
+    --create-service-linked-roles)
+      CREATE_SLR=true
+      shift # past argument
+      ;;
+    --bootstrap-region)
+      BOOTSTRAP_REGION=true
+      shift # past argument
+      ;;
+    --enable-capture-proxy)
+      ENABLE_CAPTURE_PROXY=true
+      shift # past argument
+      ;;
+    -h|--help)
+      usage
+      ;;
+    -*)
+      echo "Unknown option $1"
+      usage
+      ;;
+    *)
+      shift # past argument
+      ;;
+  esac
+done
+
+if [ "$CREATE_SLR" = true ] ; then
+  create_service_linked_roles
+fi
+
+# Store CDK context for migration solution deployment in variable
+read -r -d '' cdk_context << EOM
+{
+    "stage": "$stage",
+    "vpcId": "<VPC_ID>",
+    "engineVersion": "OS_2.9",
+    "domainName": "opensearch-cluster-aws-integ",
+    "dataNodeCount": 2,
+    "availabilityZoneCount": 2,
+    "mskBrokerNodeCount": 2,
+    "openAccessPolicyEnabled": true,
+    "domainRemovalPolicy": "DESTROY",
+    "migrationAnalyticsServiceEnabled": false,
+    "fetchMigrationEnabled": true,
+    "sourceClusterEndpoint": "<SOURCE_CLUSTER_ENDPOINT>",
+    "dpPipelineTemplatePath": "../../../test/dp_pipeline_aws_integ.yaml"
+}
+EOM
+
+git clone https://github.com/lewijacn/opensearch-cluster-cdk.git || echo "Repo already exists, skipping.."
```

Comment:
I'm guessing we're going to eventually change the location of the cdk code here.

Reply:
Not sure what our future plan will be around this, as I don't think it makes sense to have an official repo for it right now but would like to have it linked to the team somehow. For now I have gone ahead and given team members collaborator access  to this repo to make changes as needed